go/libraries/doltcore/sqle/dolt_session.go

```diff
@@ -622,6 +643,25 @@ func (sess *DoltSession) AddDB(ctx *sql.Context, db sql.Database, dbData env.DbD
 	// After setting the initial root we have no state to commit
 	sess.dirty[db.Name()] = false
 
+	err = sess.createTemporaryTablesRoot(ctx, db.Name(), ddb)
```

Comment:
This is expensive, it does IO

Rather than do this on session init, do it on demand the first time that a temp table is created

Don't maintain a temp root for sessions with no temp tables